enclave/src/pendle/math.rs

```diff
@@ -0,0 +1,327 @@
+// from here https://github.com/pendle-finance/pendle-core-v2-public/blob/46d13ce4168e8c5ad9e5641dd6380fea69e48490/contracts/core/libraries/math/LogExpMath.sol
+
+use crate::eth::aliases::*;
+
+
+pub fn ln(a: I256) -> I256 {
+    assert!(a > I256::ZERO, "ln is not defined for negative numbers and zero");
+    let one_17: I256 = "100_000_000_000_000_000".parse::<I256>().unwrap();
+    let one_18: I256 = "1_000_000_000_000_000_000".parse::<I256>().unwrap();
+
+    let ln_36_lower_bound: I256 = one_18 - one_17;
+    let ln_36_upper_bound: I256 = one_18 + one_17;
+
+    if ln_36_lower_bound < a && a < ln_36_upper_bound {
+        return _ln_36(a) / one_18;
+    } else {
+        return _ln(a);
+    }
+}
+
+
+fn _ln_36(x_inp: I256) -> I256 {
+    let one_18: I256 = "1_000_000_000_000_000_000".parse::<I256>().unwrap();
```

Comment:
I would love to see something like
```suggestion
    let one_18 = I256::from(1_000_000_000_000_000_000u128);
    // or
    let one_18: I256 = 1_000_000_000_000_000_000i128.into();
```